update basic latex tools from current tlnet

This includes important fixes:
* multicol.dtx: In \end{multicols*} only add vertical space if not
  doing already doing \raggedcolumns and add \null to hide the final
  fill if added so that it doesn't get removed again.

* multicol.dtx: change ragged columns to not just fill the
  column but also back up by \maxdepth if they otherwise become
  overfull as adding glue at the bottom means the column will
  never have any depth and before they had. (pr/4395)

  \maxdepth is applied to all column boxes. That was initially not
  necessary as \splitmaxdepth was doing the trick. However, the
  moment those boxes get reboxed this information is lost and
  resulted in some subtle errors. (pr/4395)

  Suppress even really bad overfull warnings while looping

Plus fixes for typos in the documentation

--- texlive-base-jessie.git.orig/texmf-dist/doc/latex/tools/00readme.txt
+++ /dev/null
@@ -1,120 +0,0 @@
-% \iffalse meta-comment
-%
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009
-% The LaTeX3 Project and any individual authors listed elsewhere
-% in this file.
-%
-% This file is part of the Standard LaTeX `Tools Bundle'.
-% -------------------------------------------------------
-%
-% It may be distributed and/or modified under the
-% conditions of the LaTeX Project Public License, either version 1.3c
-% of this license or (at your option) any later version.
-% The latest version of this license is in
-%    http://www.latex-project.org/lppl.txt
-% and version 1.3c or later is part of all distributions of LaTeX
-% version 2005/12/01 or later.
-%
-% The list of all files belonging to the LaTeX `Tools Bundle' is
-% given in the file `manifest.txt'.
-%
-% \fi
-
-README for the `tools' bundle  (December 1999)
-=============================
-
-This `bundle' consists of LaTeX2e packages written and supported by
-members of the LaTeX3 Project Team.
-
-The documented source code of each package is in a file with extension
-`.dtx'.  Running LaTeX on the file tools.ins will produce all the
-package files, and some associated files.
-
-So you should first process tools.ins:
-
-  latex tools.ins
-
-The files with extensions `.sty' and `.tex' (including a file whose
-name is just `.tex') should then be moved to a directory on LaTeX's
-standard input path.
-
-See the Note at the end of this file if you have problems processing
-the tools.ins file.
-
-Documentation for the individual packages may then be obtained by
-running LaTeX on the `.dtx' files.
-
-For example:
-
-  latex array.dtx
-
-will produce the file array.dvi, documenting the array package.
-
-
-The file manifest.txt contains a list of the main files in the
-distribution together with a one-or-two line summary of each package.
-
-
-Copyright
-=========
-Copyright is maintained on each of these packages by the author(s)
-of the package.
-
-
-Distribution Conditions
-=======================
-
-All the files in this bundle may be distributed under the conditions
-of the LaTeX Project Public License, either version 1.1 of this
-license or (at your option) any later version.  The latest version of
-this license is in
-   http://www.latex-project.org/lppl.txt
-and version 1.1 or later is part of all distributions of LaTeX
-version 1999/06/01 or later.
-
-The list of all files belonging to the `Tools Bundle' is
-given in the file `manifest.txt'.
-
-Commercial users of the multicol package are asked to read the
-notice at the head of the file multicol.dtx.
-
-The use of these files is otherwise unrestricted.
-
-
-Reporting Bugs
-==============
-
-If you wish to report a problem or bug in any of these packages, use
-the latexbug.tex program that comes with the standard LaTeX
-distribution.  Please ensure that you enter `tools' category when
-prompted with a menu of categories, so that the message will be
-automatically forwarded to the appropriate part of our database.
-
-When reporting bugs, please produce a small test file that shows the
-problem, and ensure that you are using the current version of the
-package, and of the base LaTeX software.
-
-
-NOTE (docstrip version)
-=======================
-
-If   latex tools.ins
-produces the `docstrip interactive mode' prompt:
-
-  * First type the extension of your input file(s):  *
-  \infileext=
-
-Then your version of docstrip is too old.
-Quit (eg by hitting `enter' to all questions) and get a newer
-docstrip.tex. It must be at least version 2.4.
-
-A suitable docstrip.tex may be found from `CTAN' archives such as
-ftp.dante.de   tex-archive/macros/latex/unpacked/docstrip.tex
-
-Docstrip is part of the base LaTeX distribution, so if you have
-an old docstrip then your LaTeX is out of date and you may consider
-getting the whole of that directory and re-installing LaTeX.
-However you need to fetch only the file docstrip.tex to unpack
-this tools distribution with your existing format.
-
--- /dev/null
+++ texlive-base-jessie.git/texmf-dist/doc/latex/tools/README
@@ -0,0 +1,120 @@
+% \iffalse meta-comment
+%
+% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
+% 2006 2008 2009
+% The LaTeX3 Project and any individual authors listed elsewhere
+% in this file.
+%
+% This file is part of the Standard LaTeX `Tools Bundle'.
+% -------------------------------------------------------
+%
+% It may be distributed and/or modified under the
+% conditions of the LaTeX Project Public License, either version 1.3c
+% of this license or (at your option) any later version.
+% The latest version of this license is in
+%    http://www.latex-project.org/lppl.txt
+% and version 1.3c or later is part of all distributions of LaTeX
+% version 2005/12/01 or later.
+%
+% The list of all files belonging to the LaTeX `Tools Bundle' is
+% given in the file `manifest.txt'.
+%
+% \fi
+
+README for the `tools' bundle  (December 1999)
+=============================
+
+This `bundle' consists of LaTeX2e packages written and supported by
+members of the LaTeX3 Project Team.
+
+The documented source code of each package is in a file with extension
+`.dtx'.  Running LaTeX on the file tools.ins will produce all the
+package files, and some associated files.
+
+So you should first process tools.ins:
+
+  latex tools.ins
+
+The files with extensions `.sty' and `.tex' (including a file whose
+name is just `.tex') should then be moved to a directory on LaTeX's
+standard input path.
+
+See the Note at the end of this file if you have problems processing
+the tools.ins file.
+
+Documentation for the individual packages may then be obtained by
+running LaTeX on the `.dtx' files.
+
+For example:
+
+  latex array.dtx
+
+will produce the file array.dvi, documenting the array package.
+
+
+The file manifest.txt contains a list of the main files in the
+distribution together with a one-or-two line summary of each package.
+
+
+Copyright
+=========
+Copyright is maintained on each of these packages by the author(s)
+of the package.
+
+
+Distribution Conditions
+=======================
+
+All the files in this bundle may be distributed under the conditions
+of the LaTeX Project Public License, either version 1.1 of this
+license or (at your option) any later version.  The latest version of
+this license is in
+   http://www.latex-project.org/lppl.txt
+and version 1.1 or later is part of all distributions of LaTeX
+version 1999/06/01 or later.
+
+The list of all files belonging to the `Tools Bundle' is
+given in the file `manifest.txt'.
+
+Commercial users of the multicol package are asked to read the
+notice at the head of the file multicol.dtx.
+
+The use of these files is otherwise unrestricted.
+
+
+Reporting Bugs
+==============
+
+If you wish to report a problem or bug in any of these packages, use
+the latexbug.tex program that comes with the standard LaTeX
+distribution.  Please ensure that you enter `tools' category when
+prompted with a menu of categories, so that the message will be
+automatically forwarded to the appropriate part of our database.
+
+When reporting bugs, please produce a small test file that shows the
+problem, and ensure that you are using the current version of the
+package, and of the base LaTeX software.
+
+
+NOTE (docstrip version)
+=======================
+
+If   latex tools.ins
+produces the `docstrip interactive mode' prompt:
+
+  * First type the extension of your input file(s):  *
+  \infileext=
+
+Then your version of docstrip is too old.
+Quit (eg by hitting `enter' to all questions) and get a newer
+docstrip.tex. It must be at least version 2.4.
+
+A suitable docstrip.tex may be found from `CTAN' archives such as
+ftp.dante.de   tex-archive/macros/latex/unpacked/docstrip.tex
+
+Docstrip is part of the base LaTeX distribution, so if you have
+an old docstrip then your LaTeX is out of date and you may consider
+getting the whole of that directory and re-installing LaTeX.
+However you need to fetch only the file docstrip.tex to unpack
+this tools distribution with your existing format.
+
--- texlive-base-jessie.git.orig/texmf-dist/doc/latex/tools/changes.txt
+++ texlive-base-jessie.git/texmf-dist/doc/latex/tools/changes.txt
@@ -5,6 +5,34 @@
 are not part of the distribution.
 =======================================================================
 
+2014-10-28    <Frank.Mittelbach@latex-project.org>
+
+	* multicol.dtx: In \end{multicols*} only add vertical space if not
+	doing already doing \raggedcolumns and add \null to hide the final
+	fill if added so that it doesn't get removed again.
+
+2014-10-28    <Frank.Mittelbach@latex-project.org>
+	* afterpage.dtx, array.dtx, bm.dtx, calc.dtx, dcolumn.dtx,
+	delarray.dtx, enumerate.dtx, ftnright.dtx, hhline.dtx, layout.dtx,
+	longtable.dtx, multicol.dtx, showkeys.dtx, tabularx.dtx,
+	theorem.dtx, trace.dtx, varioref.dtx, verbatim.dtx, xspace.dtx
+
+	Fixed typos found by Karl Berry in documentation (no version number change)
+
+2014-09-12    <Frank.Mittelbach@latex-project.org>
+
+	* multicol.dtx: change ragged columns to not just fill the
+	column but also back up by \maxdepth if they otherwise become
+	overfull as adding glue at the bottom means the column will
+	never have any depth and before they had. (pr/4395)
+
+	\maxdepth is applied to all column boxes. That was initially not
+	necessary as \splitmaxdepth was doing the trick. However, the
+	moment those boxes get reboxed this information is lost and
+	resulted in some subtle errors. (pr/4395)
+
+	Suppress even really bad overfull warnings while looping
+
 2014-08-24    <Frank.Mittelbach@latex-project.org>
 
 	* multicol.dtx: set \prevdepth to 0pt (in the right place)
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/afterpage.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/afterpage.dtx
@@ -1,7 +1,7 @@
 % \iffalse meta-comment
 %
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009
+% Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -25,7 +25,7 @@
 %
 %<package>\NeedsTeXFormat{LaTeX2e}
 %<package>\ProvidesPackage{afterpage}
-%<package>         [1995/10/27 v1.08 After-Page Package (DPC)]
+%<package>         [2014/10/28 v1.08 After-Page Package (DPC)]
 %
 %<*driver>
 \documentclass{ltxdoc}
@@ -60,7 +60,7 @@
 % \CheckSum{269}
 %
 % This package implements a command, |\afterpage|, that causes the
-% commands specified in its argument to be expanded after the curent
+% commands specified in its argument to be expanded after the current
 % page is output.\footnote{This is really a pre-release, to see whether
 % people like the idea of a command like this. This implementation is
 % \emph{not} particularly robust. This implementation does not work in
@@ -96,7 +96,7 @@
 % use one of:\\
 % |\afterpage{\clearpage\input{ltfile}}|\\
 % |\afterpage{\clearpage\input{ltfile}\clearpage}|.\\
-% The first form lets text apear on the same page as the end of the
+% The first form lets text appear on the same page as the end of the
 % longtable, the second ensures that the surrounding text starts again
 % on a new page.
 % \end{enumerate}
@@ -157,7 +157,7 @@
     \par
   \else
 %    \end{macrocode}
-% The remaining (even worse) possibility that the ouput routine
+% The remaining (even worse) possibility that the output routine
 % was triggered by the start of displaymath within a paragraph.
 %
 % Come out of displaymath with |$$|, then adjust the spacing (getting
@@ -323,7 +323,7 @@
 %    \begin{macrocode}
 \def\addboxcontents#1{{%
 %    \end{macrocode}
-% Perhaps I shouldnt use grouping here, as I probably don't really want
+% Perhaps I shouldn't use grouping here, as I probably don't really want
 % to save |#1|. If it is removed, |\splittopskip| and |\splitmaxdepth|
 % would need to be restored by hand.
 %
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/array.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/array.dtx
@@ -1,7 +1,7 @@
 % \iffalse meta-comment
 %
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009
+% Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -26,7 +26,7 @@
 %% Copyright (C) 1989-1998 Frank Mittelbach, all rights reserved.
 %<+package>\NeedsTeXFormat{LaTeX2e}[1995/06/01]
 %<+package>\ProvidesPackage{array}
-%<+package>         [2008/09/09 v2.4c Tabular extension package (FMi)]
+%<+package>         [2014/10/28 v2.4c Tabular extension package (FMi)]
 %
 % \fi
 %
@@ -129,7 +129,7 @@
 % \changes{v1.9f}{1988/05/20}{Small changes finally carried out:
 %                        1) `par!=`@empty.
 %                        2) \{..ifnum0!=!`\}...  $\to$ `bgroup and
-%                           analoguously `egroup.}
+%                           analogously `egroup.}
 % \changes{v1.9g}{1988/02/24}{Inserted again \{..ifnum0!=!`\}..,
 %                        c.f. Appendix D of the \protect\TeX{}book.}
 % \changes{v1.9h}{1988/06/28}{No longer necessary to read in the file
@@ -534,7 +534,7 @@
 % \end{enumerate}
 % \texttt{array.sty} implements the second possibility while the
 % default implementation in the \LaTeX{} kernel implements the first
-% concept. Both concepts have their merrits but one has to be aware of
+% concept. Both concepts have their merits but one has to be aware of
 % the individual implications.
 % \begin{itemize}
 % \item
@@ -837,7 +837,7 @@
 %    \texttt{latex.tex},
 %    which is the reason why the following three lines of code are
 %    commented out.
-%    Later throughout the text I will not mention it again explicitely
+%    Later throughout the text I will not mention it again explicitly
 %    whenever I use a =%= sign. These parts are already defined in
 %    \texttt{latex.tex}.
 %    \begin{macrocode}
@@ -1167,7 +1167,7 @@
 % \texttt{r}, \texttt{p}, \texttt{m} or \texttt{b} in the user
 % preamble?  In this case we have the problem of the \textsf{token}
 % register from =>{..}= and =<{..}= having to be inserted at this
-% moment because formating instructions like =\hfil= have to be set
+% moment because formatting instructions like =\hfil= have to be set
 % around them. On the other hand it is not known yet, if any =<{..}=
 % instruction will appear in the user preamble at all.
 %
@@ -1497,7 +1497,7 @@
    \prepnext@tok
 %    \end{macrocode}
 %    Now the preamble must be extended with the column whose format
-%    can be determinated by =\@chnum=.
+%    can be determined by =\@chnum=.
 %    \begin{macrocode}
    \@addtopreamble{\ifcase \@chnum
 %    \end{macrocode}
@@ -1763,7 +1763,7 @@
 %    \end{macrocode}
 %    The argument is the width of the box. This information has to be
 %    assigned to =\hsize=.
-%    Then we assain default values to several parameters used in a
+%    Then we assign default values to several parameters used in a
 %    \textsf{parbox}.
 % \changes{v2.3k}{1998/05/12}{Use \cs{setlength} to set \cs{hsize},
 %      so that the calc package can be applied here (pr/2793)}
@@ -1873,7 +1873,7 @@
 %    not surface because there =\protect= is set (which is not necessary in this
 %    implementation as there is no arbitrary user input that can get expanded) and
 %    the experimental code made the redefinition robust. Whether this is the right
-%    approach is open to question; consider the =\noexpand= a curtesy to allow an
+%    approach is open to question; consider the =\noexpand= a courtesy to allow an
 %    unsupported redefinition of a \TeX{} primitive for the moment (as people rely
 %    on that experimental code).
 %    \begin{macrocode}
@@ -2194,7 +2194,7 @@
 %    \begin{macrocode}
 \def\tabular{\gdef\@halignto{}\@tabular}
 %    \end{macrocode}
-%     and analoguesly for the star form. We evalute the argument first
+%     and analogously for the star form. We evaluate the argument first
 %     using =\setlength= so that users of the \texttt{calc} package can
 %     write code like\\ =\begin{tabular*}{(\columnwidth-1cm)/2}...=
 % \changes{v2.3l}{1998/05/13}{Use \cs{setlength} evaluate arg
@@ -2352,7 +2352,7 @@
     {\PackageWarning{array}{Column \NC@char\space is already defined}}%
 %    \end{macrocode}
 % Now we define a macro with an argument delimited by the new column
-% specifier, this is used to find occurences of this specifier in the
+% specifier, this is used to find occurrences of this specifier in the
 % user preamble.
 %    \begin{macrocode}
   \@namedef{NC@find@\NC@char}##1#1{\NC@{##1}}%
@@ -2378,7 +2378,7 @@
 %    \end{macrocode}
 % \end{macro}
 % \begin{macro}{\NC@}
-% Having found an occurence of the new column, save the preamble
+% Having found an occurrence of the new column, save the preamble
 % before the column in "\@temptokena", then check to see if we
 % are at the end of the preamble. (A dummy occurrence of the column
 % specifier will be placed at the end of the preamble by "\NC@do".
@@ -2440,7 +2440,7 @@
 % definition of "\NC@rewrite@C". Actually as the the list always
 % starts off with "\NC@do *" and we do not want to print the
 % definition of the $*$-form, define "\NC@do" to throw away the first
-% item in the list, and then redefine itsef to print the rest of the
+% item in the list, and then redefine itself to print the rest of the
 % definitions.
 %    \begin{macrocode}
 \def\showcols{{\def\NC@do##1{\let\NC@do\NC@show}\the\NC@list}}
@@ -2692,7 +2692,7 @@
 %             \texttt{\textbackslash lasthline}}
 %
 % The Companion~\cite[p.137]{bk:GMS94} suggests two additional
-% commands to control the allignments in case of tabulars with
+% commands to control the alignments in case of tabulars with
 % horizontal lines. They are now added to this package.
 %
 %  \begin{macro}{\extratabsurround}
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/bm.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/bm.dtx
@@ -1,7 +1,7 @@
 % \iffalse meta-comment
 %
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009
+% Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -36,7 +36,7 @@
 %<driver>\ProvidesFile{bm.drv}
 % \fi
 %         \ProvidesFile{bm.dtx}
-          [2004/02/26 v1.1c Bold Symbol Support (DPC/FMi)]
+          [2014/10/28 v1.1c Bold Symbol Support (DPC/FMi)]
 %
 % \iffalse
 %<*driver>
@@ -266,7 +266,7 @@
 % bold weights, then you would declare this to \LaTeX\ via:\\
 % |\DeclareSymbolFont{extras}  {OMS}{xyz}{m}{n}|\\
 % |\SetSymbolFont{extras}{bold}{OMS}{xyz}{bx}{n}|\\
-% At this point the symbols wil be available in the normal math
+% At this point the symbols will be available in the normal math
 % version, and their bold variants in |\boldmath|. If you also
 % declare:\\
 % |\DeclareSymbolFont{boldextras}{OMS}{xyz}{bx}{n}|\\
@@ -290,7 +290,7 @@
 % for instance, for a relation, |\bm{\mathrel{\cmd}}|.
 %
 % \subsection{AMS package \textsf{amsbsy}}
-% The |\bm| command shares some functionailty with the |\boldsymbol|
+% The |\bm| command shares some functionality with the |\boldsymbol|
 % command from the AMS \LaTeX\ collection. To aid in moving documents
 % between these two packages, this package defines |\boldsymbol| and
 % |\heavysymbol| as alternative names for |\bm| and |\hm|.
@@ -625,7 +625,7 @@
 % To save declaring too many symbol fonts, do not auto-declare
 % any more than |\bmmax| bold symbol fonts into the normal math
 % version. Any bold fonts not so allocated will be accessed via
-% |\boldmath| which is slower and doesnt work for delimiters
+% |\boldmath| which is slower and doesn't work for delimiters
 % and accents. It may be set in the preamble with |\newcommand|
 % but use |\chardef| here for a slight efficiency gain.
 %
@@ -1269,7 +1269,7 @@
 % \changes{v0.07}{1996/12/14}
 %      {Combine all register macros}
 % |\mkern| itself would transfer to |\bm@command| without any special
-% test, but any explict dimension following would be converted to
+% test, but any explicit dimension following would be converted to
 % |\mathchar|. So trap this and grab the muskip as a muskip.
 % This is used in |\iiint|. |\penalty| was needed for the
 % AMS  version of |\colon|, and so do most of the others as well.
@@ -1461,7 +1461,7 @@
 %
 % It would be possible to locally make mathaccents take an argument
 % during the bm look-ahead, so the brace groups would then vanish
-% during expansion, however I would then need to explitly skip past
+% during expansion, however I would then need to explicitly skip past
 % \meta{filler} and also make sure that the end of parse token
 % was not gobbled in marginal cases like |$\bm\hat$|.
 %
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/calc.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/calc.dtx
@@ -1,7 +1,7 @@
 % \iffalse meta-comment
 %
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009
+% Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -40,7 +40,7 @@
 %<driver> \ProvidesFile{calc.drv}
 % \fi
 %         \ProvidesFile{calc.dtx}
-          [2007/08/22 v4.3 Infix arithmetic (KKT,FJ)]
+          [2014/10/28 v4.3 Infix arithmetic (KKT,FJ)]
 %
 % \iffalse
 %<*driver>
@@ -811,7 +811,7 @@
 %    These do not need any particular definition when they are scanned
 %    so, for efficiency and robustness, we make them all equivalent to
 %    the same harmless (I hope) unexpandable command.\footnote{If this
-%    level of safety is not needed then the code can be speeded up:
+%    level of safety is not needed then the code can be sped up:
 %    CAR.}  Thus the test in |\@calc@pre@scan| finds any of them.
 %
 %    As we have to check for these commands explicitly we must ensure
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/dcolumn.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/dcolumn.dtx
@@ -1,7 +1,7 @@
 % \iffalse meta-comment
 %
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009
+% Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -31,7 +31,7 @@
 %<driver>\ProvidesFile{dcolumn.drv}
 % \fi
 %         \ProvidesFile{dcolumn.dtx}
-          [2001/05/28 v1.06 decimal alignment package (DPC)]
+          [2014/10/28 v1.06 decimal alignment package (DPC)]
 %
 % \iffalse
 %<*driver>
@@ -215,7 +215,7 @@
 % \item The separator is not given its special definition by making it
 % active, as this would not work for an entry such as "& .5 &", as the
 % first token of an alignment entry is read {\em before\/} the preamble
-% part, incase it is an "\omit", in which case the preamble is to be
+% part, in case it is an "\omit", in which case the preamble is to be
 % omitted. Instead we switch the mathcode to (hex) 8000, which makes the
 % token act as if it were active.
 % \item Although \verb|\mathcode`.="8000|  makes {\tt.} act as if it
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/delarray.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/delarray.dtx
@@ -1,7 +1,7 @@
 % \iffalse meta-comment
 %
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009
+% Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -25,7 +25,7 @@
 %
 %<package>\NeedsTeXFormat{LaTeX2e}
 %<package>\ProvidesPackage{delarray}
-%<package>         [1994/03/14 v1.01 array delimiter package (DPC)]
+%<package>         [2014/10/28 v1.01 array delimiter package (DPC)]
 %
 %<*driver>
 \documentclass{ltxdoc}
@@ -139,7 +139,7 @@
 %    \end{macrocode}
 % The following line is completely redundant but it does catch errors
 % involving delimiters  before the processing of the alignment begins.
-% A common error is likely to be omiting the `.' in a
+% A common error is likely to be omitting the `.' in a
 % |\cases|-type construction. This causes the first token of the
 % alignment to be gobbled, possibly causing lots of spurious errors
 % before the cause of the error, the missing delimiter, is discovered as
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/enumerate.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/enumerate.dtx
@@ -1,7 +1,7 @@
 % \iffalse meta-comment
 %
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009
+% Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -31,7 +31,7 @@
 %<driver> \ProvidesFile{enumerate.drv}
 % \fi
 %         \ProvidesFile{enumerate.dtx}
-          [1999/03/05 v3.00 enumerate extensions (DPC)]
+          [2014/10/28 v3.00 enumerate extensions (DPC)]
 %
 % \iffalse
 %<*driver>
@@ -65,7 +65,7 @@
 % This package gives the enumerate environment an optional argument
 % which determines the style in which the counter is printed.
 %
-% An occurence of one of the tokens |A a I i| or |1| produces the value
+% An occurrence of one of the tokens |A a I i| or |1| produces the value
 % of the counter printed with (respectively) |\Alph \alph \Roman \roman|
 % or |\arabic|.
 %
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/ftnright.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/ftnright.dtx
@@ -1,7 +1,7 @@
 % \iffalse meta-comment
 %
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009 2010
+% Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -26,7 +26,7 @@
 %% Copyright (C) 1989-2004 Frank Mittelbach, all rights reserved.
 %<+package>\NeedsTeXFormat{LaTeX2e}[1995/06/01]
 %<+package>\ProvidesPackage{ftnright}
-%<+package>         [2010/02/25 v1.1f footnote layout package (FMi)]
+%<+package>         [2014/10/28 v1.1f footnote layout package (FMi)]
 %
 % \fi
 %%
@@ -332,7 +332,7 @@
 % \bibitem{src:ltxiii94}
 % \LaTeX3 project.
 % \newblock \LaTeXe distribution, 1994.
-% \newblock Souces for {\LaTeXe} the successor to \LaTeX~2.09.
+% \newblock Sources for {\LaTeXe} the successor to \LaTeX~2.09.
 %
 % \bibitem{art:FMi89b}
 % Frank Mittelbach
@@ -526,7 +526,7 @@
   \global\@firstcolumnfalse
 %    \end{macrocode}
 %    After changing the switch, we save the first column (which was
-%    placed by preceeding macros in |\@outputbox|) in the box register
+%    placed by preceding macros in |\@outputbox|) in the box register
 %    |\@leftcolumn|. Since we are inside the output routine, all those
 %    assignments have to be |\global| to take any effect.
 %    \begin{macrocode}
@@ -548,7 +548,7 @@
 %
 %    If we have just finished the right column, i.e., when
 %    |\if@firstcolumn| equals |\iffalse|, we will reset the |\footins|
-%    parameters as explained above using the utitlity macro
+%    parameters as explained above using the utility macro
 %    |\preparefootins|.
 %    \begin{macrocode}
  \else \preparefootins
@@ -673,7 +673,7 @@
 %    \end{macrocode}
 %    We then reduce the |\@colht| by this amount and again assign
 %    |\@colroom| the value of |\@colht|. If no footnotes are present,
-%    we substract zero, so there is no harm in doing this operation
+%    we subtract zero, so there is no harm in doing this operation
 %    all the time.
 %    \begin{macrocode}
    \global\advance\@colht-\ftn@amount
@@ -774,7 +774,7 @@
   \setbox\@outputbox\box\@cclv
 %    \end{macrocode}
 %
-%    But if the user errously forgot to specify a twocolumn layout, we
+%    But if the user erroneously forgot to specify a twocolumn layout, we
 %    will always typeset the first column, so that the footnotes are
 %    never printed.  Therefore we better check for this special case
 %    and output the footnotes on a separate page in an
@@ -847,7 +847,7 @@
 %    package.\footnote{This decision is certainly open to criticism,
 %    since there is nothing to separate. On the other hand, a rule or
 %    some other ornament in front of the footnotes is part of the
-%    design which should be used concistently throughout a document.
+%    design which should be used consistently throughout a document.
 %    As a last argument in favor of the rule, consider the situation
 %    where \LaTeX{} decided to place only floats and footnotes into
 %    the right hand column. In this case a separator again seems
@@ -882,7 +882,7 @@
 %    shorter, only a cosmetic change. The use of this hardwired {\sf
 %    dimen} register seems to indicate that this part of \LaTeX{} was
 %    written very early and left unchanged since then: an interesting
-%    fact for software archeologists.}\footnote{For the \LaTeXe{}
+%    fact for software archaeologists.}\footnote{For the \LaTeXe{}
 %    upgrade I had to add the support for the
 %    {\tt\string\enlargethispage} command---let's hope I did it in the
 %    correct way.}
@@ -969,7 +969,7 @@
 %
 % \begin{macro}{\footnoterule}
 %    Setting footnotes in smaller type and separating them with
-%    sufficiant space from the main text allow us to omit the
+%    sufficient space from the main text allow us to omit the
 %    |\footnoterule| normally used.
 %    \begin{macrocode}
 \let\footnoterule\@empty
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/hhline.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/hhline.dtx
@@ -1,7 +1,7 @@
 % \iffalse meta-comment
 %
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009
+% Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -25,7 +25,7 @@
 %
 %<package>\NeedsTeXFormat{LaTeX2e}
 %<package>\ProvidesPackage{hhline}
-%<package>         [1994/05/23 v2.03 Table rule package (DPC)]
+%<package>         [2014/10/28 v2.03 Table rule package (DPC)]
 %
 %<*driver>
 \documentclass{ltxdoc}
@@ -91,7 +91,7 @@
 % \]
 % If a double vline is specified ("||" or "::") then the hlines
 % produced by "\hhline" are broken. To obtain the effect of an hline
-% `cutting through'the double vline, use a "#" or omit the vline
+% `cutting through' the double vline, use a "#" or omit the vline
 % specifiers, depending on whether or not you wish the double vline to
 % break.
 %
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/layout.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/layout.dtx
@@ -1,7 +1,7 @@
 % \iffalse meta-comment
 %
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009
+% Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -31,7 +31,7 @@
 %<+package>\ProvidesPackage{layout}
 %<+driver>\ProvidesFile{layout.drv}
 %\ProvidesFile{layout.dtx}
-                [2000/09/25 v1.2c Show layout parameters]
+                [2014/10/28 v1.2c Show layout parameters]
 %
 %    A short driver is provided that can be extracted if necessary by
 %    the \textsf{DocStrip} program provided with \LaTeXe.
@@ -759,7 +759,7 @@
     \advance\PositionY by \ref@body
 %    \end{macrocode}
 %
-%    The x-psition of the arrow is at $4/5$ of the width of the `body
+%    The x-position of the arrow is at $4/5$ of the width of the `body
 %    box'.
 %    \begin{macrocode}
     \PositionX = \cnt@textwidth
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/longtable.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/longtable.dtx
@@ -1,7 +1,7 @@
 % \iffalse meta-comment
 %
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009
+% Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -31,7 +31,7 @@
 %<driver> \ProvidesFile{longtable.drv}
 % \fi
 %         \ProvidesFile{longtable.dtx}
-          [2004/02/01 v4.11 Multi-page Table package (DPC)]
+          [2014/10/28 v4.11 Multi-page Table package (DPC)]
 %
 % \iffalse
 %<*driver>
@@ -82,7 +82,7 @@
 %
 % \changes{v3.01}{1992/04/06}
 %    {(Michel Goossens) If a chunk ends on a line in which the first
-%     entry is empty, wierd errors occur. Added special begin and end
+%     entry is empty, weird errors occur. Added special begin and end
 %     groups, (\cs{ifnum}0 ) stuff as  explained in Appendix D.}
 %
 % \changes{v3.02}{1992/04/09}
@@ -500,11 +500,11 @@
 % accordingly. If this has to propagate to chunks before the
 % "\multicolumn" one, an additional pass will, of course, be
 % needed. It is possible to construct tables where this rippling up of
-% the correct widths taks several passes to `converge' and produce a
+% the correct widths takes several passes to `converge' and produce a
 % table with all chunks aligned. However in
 % order to need many passes one needs to construct a table with
 % many overlapping "\multicolumn" entries, all being wider than the
-% natural widths of the columns they span, and all occuring in
+% natural widths of the columns they span, and all occurring in
 % different chunks. In the typical case the algorithm will converge
 % after three or four passes, and, the benefits of not needing to edit
 % the document before the final run to add "\setlongtables", and the
@@ -854,7 +854,7 @@
 \ProcessOptions
 %    \end{macrocode}
 %
-% \subsection{User Setable Parameters}
+% \subsection{User Settable Parameters}
 %
 % \begin{macro}{\LTleft}
 % \begin{macro}{\LTright}
@@ -1059,7 +1059,7 @@
 %      {(Mike Van Geest) rename \cs{LT@mc} to
 %         \cs{LT@mcol} to allow 1100 tables}
 % \changes{v4.01}{1996/04/11}
-%      {(DPC) dont need multicolumn warning}
+%      {(DPC) don't need multicolumn warning}
 %\begin{verbatim}
 %  \global\let\LT@mcw@rn\relax
 %\end{verbatim}
@@ -1480,7 +1480,7 @@
 % \end{macro}
 %
 % \begin{macro}{\LT@xtabularcr}
-% This justs looks for an optional argument.
+% This just looks for an optional argument.
 %    \begin{macrocode}
 \def\LT@xtabularcr{%
   \@ifnextchar[\LT@argtabularcr\LT@ntabularcr}
@@ -1539,7 +1539,7 @@
 %
 % Previous releases left the "\lineskip" glue at the end of each chunk
 % that had been added when the dummy row was added. There was no glue
-% at the start of the next chunk as \TeX\ normally dooes not put
+% at the start of the next chunk as \TeX\ normally does not put
 % "\lineskip" glue at the top of a box. This meant that normally the
 % chunks fitted together perfectly, however "\noalign" material at a
 % chunk boundary came before the first row of the next chunk
@@ -1647,7 +1647,7 @@
 % \changes{v4.00}{1996/04/08}
 %      {(DK) Macro added}
 % \changes{v4.01}{1996/04/11}
-%      {(DPC) Dont mess with \cs{multicolumn} on draft passes}
+%      {(DPC) Don't mess with \cs{multicolumn} on draft passes}
 % Create a blank row if we are not using the info in the ".aux" file.
 %    \begin{macrocode}
 \def\LT@blank@row{%
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/multicol.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/multicol.dtx
@@ -69,7 +69,7 @@
 %%   - multicol is used for non-commercial purposes
 %%
 %%
-%% Determinating a license fee might result in a license fee of zero
+%% Determining a license fee might result in a license fee of zero
 %% (i.e., no payment) in case a producer has determined that the use
 %% of multicol has no enhancing effect on the product. This is a
 %% plausible scenario, i.e., in the above two cases the producer is
@@ -91,10 +91,10 @@
 %<driver> \ProvidesFile{multicol.drv}
 % \fi
 %         \ProvidesFile{multicol.dtx}
-          [2014/08/24 v1.8g  multicolumn formatting (FMi)]
+          [2014/10/28 v1.8i  multicolumn formatting (FMi)]
 %
 %
-%% \CheckSum{1840}
+%% \CheckSum{1853}
 %% \CharacterTable
 %%  {Upper-case    \A\B\C\D\E\F\G\H\I\J\K\L\M\N\O\P\Q\R\S\T\U\V\W\X\Y\Z
 %%   Lower-case    \a\b\c\d\e\f\g\h\i\j\k\l\m\n\o\p\q\r\s\t\u\v\w\x\y\z
@@ -197,7 +197,7 @@
 %    over to next page.
 %
 %    Also added was support for |\enlargethispage|. This means it is
-%    now possible to request a page to be artifically enlarged or
+%    now possible to request a page to be artificially  enlarged or
 %    shortened. Note that if you enlarge pages by more than one line
 %    you may have to increase the {\sf collectmore} counter value to
 %    ensure that enough material is being picked up.
@@ -246,7 +246,7 @@
 % \begin{multicols}{3}
 %    Right-to-left typesetting will only reverse the column orders. Any
 %    other support needed will have to be provided by other means,
-%    e.g., using apprpriate fonts and reversing the writing directions
+%    e.g., using appropriate fonts and reversing the writing directions
 %    within the columns. As footnotes are typeset in full measure the
 %    footnote rule needs to be redefined as if they are below a single
 %    column, i.e., using |\textwidth| not |\columnwidth|. For example:
@@ -267,7 +267,7 @@
 %   been added to the balancing routine. The balancing routine now
 %   checks the badness
 %   of the resulting columns and rejects solutions that are larger
-%   than a certain treshold.
+%   than a certain threshold.
 %   At the same time \mc{} has been upgraded to run under \LaTeXe{}.
 %
 %   Later changes to 1.5 include |\columnbreak| and |multicols*|.
@@ -621,7 +621,7 @@
 %  inside `multicols' environment!}
 %
 %  This message appears if you try to use the |\marginpar| command or
-%  an unstared version of the {\sf figure} or {\sf table} environment.
+%  an unstarred version of the {\sf figure} or {\sf table} environment.
 %  Such floats will disappear!
 %
 % \item[] {\hspace*{-\labelsep}\tt Very deep columns!
@@ -938,7 +938,7 @@
 %<*package>
 % \NeedsTeXFormat{LaTeX2e}
 % \ProvidesPackage{multicol}[..../../..
-%    v... multicolum formatting]
+%    v... multicolumn formatting]
 %    \end{macrocode}
 %
 %^^A \subsection{Option processing}
@@ -1451,7 +1451,7 @@
 %    \begin{macrocode}
     \advance\@colroom-\ht\partial@page
 %    \end{macrocode}
-%    Then we have to calulate the |\vsize| value to use during column
+%    Then we have to calculate the |\vsize| value to use during column
 %    assembly. |\set@mult@vsize| takes an argument which allows to
 %    make the setting local (|\relax|) or global (|\global|). The
 %    latter variant is used inside the output routine below. At this
@@ -1583,7 +1583,7 @@
 %                       has the disadvantage that the resulting
 %                       program cannot be called \TeX{} thereafter.}
 %    this simple trick saved us from rewriting the |\footnote|
-%    macros. However, some applications refered to |\columnwidth| as
+%    macros. However, some applications referred to |\columnwidth| as
 %    the ``width of the current column'' to typeset displays
 %    (the \texttt{amsmath} package, for example) and to allow the use
 %    of such applications together with \texttt{multicol} this is now
@@ -1718,7 +1718,7 @@
 %    \end{macrocode}
 %    If there was a |\columnbreak| on the very last line all material
 %    will have been moved to the |\colbreak@box|. Thus the the galley
-%    will be ampty and no output routine gets called so that the text
+%    will be empty and no output routine gets called so that the text
 %    is lost. To avoid this problem (though unlikely) we check if the
 %    current galley is empty and the |\colbreak@box| contains text and
 %    if so return that to the galley. If the galley is non-empty any
@@ -1772,7 +1772,7 @@
 %    case that the |\eject| will have no effect (since the main
 %    vertical list is empty)---thus no output routine is called at
 %    all. As a result the material preceding the \mc{} (stored in
-%    |\partial@page| will get lost if we don't take of this by hand.
+%    |\partial@page| will get lost if we don't put this back by hand.
 % \changes{v1.4m}{1992/09/04}{Check \cs{partial@page} being emptied}
 %    \begin{macrocode}
     \ifvbox\partial@page
@@ -1830,7 +1830,7 @@
 %    which can happen if there is a break point just before it (such
 %    as end ending environment) which was chosen. In that case we
 %    would do the next page using the internal |\vsize| for multicol
-%    collection which is a desaster. So we better catch this
+%    collection which is a disaster. So we better catch this
 %    case. Fortunately we can detect it by looking at |\pagegoal|.
 % \changes{v1.5x}{2000/05/05}{Detect and fix problem if a multicols
 %    ends at the top of a page}
@@ -1843,6 +1843,19 @@
   \fi
   \addvspace\multicolsep
 %    \end{macrocode}
+%    There is one more thing to do: the balanced result of the
+%   environment is supposed to have a |\prevdepth| of zero as we
+%   backed up by its real prevdepth within |\page@sofar|. However if
+%   the balancing happened in the output routine then \TeX{} reverts
+%   to the |\prevdepth| that was current before the OR once the OR has
+%   finished. In short |\prevdepth| is something you can't set
+%   globally it is alway local to the current list being built. Thus
+%   we need to set it back to zero here to avoid incorrect spacing.
+% \changes{v1.8h}{2014/09/12}{Set \cs{prevdepdth} for current vlist
+%   when returning from multicols environment}
+%    \begin{macrocode}
+ \prevdepth\z@
+%    \end{macrocode}
 %    If statistics are required we finally report that we have
 %    finished everything.
 %    \begin{macrocode}
@@ -2051,16 +2064,21 @@
 %    The depths of the columns depend on their last lines. To ensure
 %    that we will always get a similar look as far as the rules are
 %    concerned we force the depth to be at least the depth of a
-%    letter~`p' (which is what we initialized |\dimen2| to above).
+%    letter~`p' (which is what we set |\dimen2| to above).
 %    \begin{macrocode}
        \rlap{\phantom p}%
     }%
 %    \end{macrocode}
 %    The processed material might consist of a last line with a
-%    decender in which case the |\prevdepth| will be
+%    descender in which case the |\prevdepth| will be
 %    non-zero. However, this material is getting reformatted now so
 %    that this value is likely to be wrong. We therefore normalize the
-%    situation by pretending that the depth is zero.
+%    situation by pretending that the depth is zero. However, if |\page@sofar|
+%   is being called inside the OR then setting |\prevdepth| here has
+%   no longlasting effect, we therefore have to repeat this once we
+%   return to the main vertical list. Here we set it only for those
+%   cases where the command is used within a list and then followed by
+%   something else.
 % \changes{v1.8g}{2014/08/24}{Resetting \cs{prevdepth} in the right place}
 %    \begin{macrocode}
    \prevdepth\z@
@@ -2118,6 +2136,23 @@
 % \end{macro}
 %
 %
+% \begin{macro}{\vfillmaxdepth}
+%    This curious definition is used as the space at the bottom of a
+%   column if we implement |\raggedcolumns|. Normlly one only appends
+%   |\vfill| in that case but this is actually wrong for columns that
+%   are more or less full: by adding a glue at the bottom such a
+%   column doesn't have any depth any more but without it the material
+%   would be allowed a depth of |\maxdepth|. So we allow shrinking by
+%   that amount. This only makes a difference if the box would
+%   otherwise become overfull and shrinking never exceeds the
+%   specified value, so we should be fine.
+% \changes{v1.8h}{2014/09/12}{Macro added  (pr/4395)}
+%    \begin{macrocode}
+\def\vfillmaxdepth{\vskip \z@ \@plus 1fill \@minus \maxdepth}
+%    \end{macrocode}
+% \end{macro}
+%
+%
 % \begin{macro}{\multi@column@out}
 %    Now we can't postpone the difficulties any longer.  The
 %    |\multi@column@out| routine will be called in two situations.
@@ -2229,18 +2264,20 @@
 %         this was wrong and unboxing is always needed.}
 % \changes{v1.8b}{2013/08/28}{Remove discardable items at the
 %                             end of split boxes}
+% \changes{v1.8h}{2014/09/12}{Use \cs{vfillmaxdepth}}
 %    \begin{macrocode}
             \setbox\count@
                  \vbox to\dimen@
                   {\unvbox\count@
                    \remove@discardable@items
-                   \ifshr@nking\vfill\fi}%
+                   \ifshr@nking\vfillmaxdepth\fi}%
            }%
 %    \end{macrocode}
 %    Then the last column follows.
 % \changes{v1.5a}{1992/11/04}{New box mechanism}
 % \changes{v1.8b}{2013/08/28}{Remove discardable items at the
 %                             end of split boxes}
+% \changes{v1.8h}{2014/09/12}{Use \cs{vfillmaxdepth}}
 %    \begin{macrocode}
    \setbox\mult@rightbox
        \vsplit\@cclv to\dimen@
@@ -2248,7 +2285,7 @@
    \setbox\mult@rightbox\vbox to\dimen@
           {\unvbox\mult@rightbox
            \remove@discardable@items
-           \ifshr@nking\vfill\fi}%
+           \ifshr@nking\vfillmaxdepth\fi}%
 %    \end{macrocode}
 %    Now that we are done with the boxes, we restored the current
 %    setting for shrinking in case it got changed:
@@ -2258,7 +2295,7 @@
 %    Having done this we hope that box 255 is emptied. If not, we
 %    reinsert its contents.
 % \changes{v1.8a}{2011/12/20}{Only re-add output penalty if it was
-%    explictly set}
+%    explicitly set}
 %    \begin{macrocode}
    \ifvoid\@cclv \else
        \unvbox\@cclv
@@ -2360,7 +2397,7 @@
 %    \end{macrocode}
 %    After the page is shipped out we have to prepare the kept marks
 %    for the following page. |\kept@firstmark| and |\kept@botmark|
-%    reinitilized by setting them to |\@empty|.  The value of
+%    reinitialized by setting them to |\@empty|.  The value of
 %    |\botmark| is then assigned to |\kept@topmark|.
 % \changes{v1.4g}{1992/06/03}{Only change \cs{kept@topmark} if
 %                            \cs{kept@botmark} non-empty}
@@ -2378,7 +2415,7 @@
 %    \end{macrocode}
 %    Now we reset |\@colroom| to |\@colht| which is \LaTeX's
 %    saved value of |\textheight|.
-%    We also ahve to reset the recorded position of the last
+%    We also have to reset the recorded position of the last
 %   |\marginpar| as we are now on a new page.
 % \changes{v1.8d}{2014/04/23}{Reset \cs{@mparbottom} after page finishes}
 %    \begin{macrocode}
@@ -2699,7 +2736,7 @@
 %    \begin{macrocode}
    \get@keptmarks\mult@box
 %    \end{macrocode}
-%    We then contine by resetting trying to remove any discardable
+%    We then continue by resetting trying to remove any discardable
 %    stuff at the end of |\mult@box|. This is rather experimental.  We
 %    also add a forced break point at the very beginning, so that we
 %    can split the box to height zero later on, thereby adding a known
@@ -2723,6 +2760,16 @@
        \@minus\multicolovershoot
    \splitmaxdepth\maxdepth
 %    \end{macrocode}
+%   We also have to set |\boxmaxdepth| which normally allows to
+%   build boxes with arbitrary depth, but as we are building text
+%   columns we really want to restrict the depth. This is necessary as
+%   we sometimes rebox the boxes generated by |\vsplit| and then the
+%   restriction posed by |\splitmaxdepth|  gets lost.
+% \changes{v1.8h}{2014/09/12}{All column boxes should obey
+%   \cs{maxdepth}  (pr/4395)}
+%    \begin{macrocode}
+   \boxmaxdepth\maxdepth
+%    \end{macrocode}
 %    The next step is a bit tricky: when \TeX{} assembles material in
 %    a box, the first line isn't preceded by interline glue, i.e.
 %    there is no parameter like |\boxtopskip| in \TeX{}. This means
@@ -2812,9 +2859,10 @@
    \vbadness\@M
 %    \end{macrocode}
 %    We also allow for overfull boxes while we trying to split the
-%    columns.
+%    columns. They can easily happen if we have objects with unusual depth.
+% \changes{v1.8h}{2014/09/12}{Do not report overfull}
 %    \begin{macrocode}
-   \vfuzz \col@number\baselineskip
+   \vfuzz \maxdimen
 %    \end{macrocode}
 %    The variable |\last@try| will hold the dimension used in the
 %    previous trial splitting. We initialize it with a negative value.
@@ -2899,7 +2947,8 @@
                         }}%
 %    \end{macrocode}
 %    There is one subtle point here: while all other constructed boxes
-%    have a depth that is determined by |\splitmaxdepth| the last box
+%    have a depth that is determined by |\splitmaxdepth| and/or
+%   |\boxmaxdepth| the last box
 %    will get a natural depth disregarding the original setting and
 %    the value of |\splitmaxdepth| or |\boxmaxdepth|. This means that
 %    we may end up with a very large depth in box |\mult@grightbox|
@@ -2907,7 +2956,6 @@
 %    make the result of the testing incorrect. So we change the value
 %    by unboxing the box into itself.
 %    \begin{macrocode}
-   \boxmaxdepth\maxdepth
    \global\setbox\mult@grightbox
       \vbox{\unvbox\mult@grightbox}%
 %    \end{macrocode}
@@ -2991,7 +3039,7 @@
 %    If the height of the last box is too large we mark this trial as
 %    unsuccessful.
 % \changes{v1.5v}{1999/07/18}{Added tracing statements for
-%                             trial unsuccesful}
+%                             trial unsuccessful}
 %    \begin{macrocode}
 %<*badness>
       \too@badtrue
@@ -3038,11 +3086,12 @@
 %    We then compare this badness with the allowed badness for the final
 %    column. If it does not exceed this value we use the box, otherwise
 %    we rebox it once more and add some glue at the bottom.
+% \changes{v1.8h}{2014/09/12}{Use \cs{vfillmaxdepth}}
 %    \begin{macrocode}
         \ifnum\badness>\c@finalcolumnbadness
           \global\setbox\mult@grightbox
            \vbox to\dimen@
-              {\unvbox\mult@grightbox\vfill}%
+              {\unvbox\mult@grightbox\vfillmaxdepth}%
            \ifnum\c@tracingmulticols>\@ne
              \message{ setting natural
               (> \the\c@finalcolumnbadness)}%
@@ -3066,7 +3115,7 @@
 %    \end{macrocode}
 %    If the natural height of the first box is smaller than the
 %    current trial size but is larger than the previous trial size it
-%    is likely that we have missed a potientially better
+%    is likely that we have missed a potentially better
 %    solution. (This could have happened if for some reason our first
 %    trial size was too high.) In that case we dismiss this trial and
 %    restart using the natural height for the next trial.
@@ -3096,7 +3145,7 @@
 %    At that point |\dimen@| holds the height that was determined by
 %    the balancing loop.
 %    If that height for the columns turns out to be larger
-%    than the available space (which is |\@colroom|) we sqeeze the
+%    than the available space (which is |\@colroom|) we squeeze the
 %    columns into the space assuming that they will have enough
 %    shrinkability to allow this.\footnote{This might be wrong, since
 %    the shrinkability that accounts for the amount of material might
@@ -3139,12 +3188,13 @@
            {%
 %    \end{macrocode}
 %
+% \changes{v1.8h}{2014/09/12}{Use \cs{vfillmaxdepth}}
 %    \begin{macrocode}
             \vskip \z@
               \@plus-\multicolundershoot
               \@minus-\multicolovershoot
             \unvbox\@tempcnta
-            \ifshr@nking\vfill\fi
+            \ifshr@nking\vfillmaxdepth\fi
            }%
 %    \end{macrocode}
 %    If the resulting box is overfull there was too much material to
@@ -3280,7 +3330,7 @@
    \@largefloatcheck
    \outer@nobreak
 %    \end{macrocode}
-%    This is cheap (defering the floats until after the current page)
+%    This is cheap (deferring the floats until after the current page)
 %    but any other solution would go deep into \LaTeX's output
 %    routine and I don't like to work on it until I know which parts
 %    of the output routine have to be reimplemented anyway for
@@ -3510,7 +3560,7 @@
 %  \begin{macro}{\remove@discardable@items}
 %    There are situations when we may have some space at the end of a
 %    column and this macro here will attempt to get rid of it. This
-%    will not remove an extremly long sequence of spaces and
+%    will not remove an extremely long sequence of spaces and
 %    penalties, but in nearly all cases this will be enough.
 %    \begin{macrocode}
 \def\remove@discardable@items{%
@@ -3609,7 +3659,7 @@
 %    definition.
 %
 %    Starting with version v1.5r this is now done in a way that the original
-%    definition is still used, execpt that locally |\columnwidth| is set to
+%    definition is still used, except that locally |\columnwidth| is set to
 %    |\textwidth|.
 %
 %    This solves two problems: first redefinitions of |\@footnotetext|
@@ -3667,10 +3717,19 @@
 %    When ending the environment we simply end the inner
 %    \texttt{multicols} environment, except that we better also stick
 %    in some stretchable vertical glue so that the last column still
-%    containing text is not vertically stretched out.
+%    containing text is not vertically stretched out. The final |\null| hides this space
+%    so that it doesn't get removed again when the columns are assembled.
+%    However, we only do this if we aren't already doing |\raggedcolumns|
+%    because it would be a mistake to add such glue twice. After all it contains
+%    a minus component of |\maxdepth| and that would then be applied twice!
 % \changes{v1.5q}{1998/01/19}{Macro added}
-%    \begin{macrocode}
- \@namedef{endmulticols*}{\vfill
+% \changes{v1.8h}{2014/09/12}{Use \cs{vfillmaxdepth}}
+% \changes{v1.8i}{2014/10/28}{Add \cs{null} to hide the final fill and only add
+%                             vertical space if not doing \cs{raggedcolumns}}
+%    \begin{macrocode}
+ \@namedef{endmulticols*}{\ifshr@nking\else
+                             \vfillmaxdepth\nointerlineskip\null
+                          \fi
    \end{multicols}}
 %</nobalance>
 %    \end{macrocode}
@@ -4013,7 +4072,7 @@
 %\mc@set@col@status{lcol-4}{0}
 %\end{verbatim}
 %   The \verb=\mc@col@status= line denotes the column type and has been
-%   writting out just before corresponding the column box was placed
+%   written out just before corresponding the column box was placed
 %   onto the page.
 %   The\verb=\mc@set@col@status= lines have been written out as part
 %   of shipping the column boxes out, e.g.,
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/showkeys.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/showkeys.dtx
@@ -1,7 +1,7 @@
 % \iffalse meta-comment
 %
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009 2011
+% Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -32,7 +32,7 @@
 %<driver> \ProvidesFile{showkeys.drv}
 % \fi
 %         \ProvidesFile{showkeys.dtx}
-          [2014/04/24 v3.17 Show cite and label keys (DPC, MH)]
+          [2014/10/28 v3.17 Show cite and label keys (DPC, MH)]
 %
 % \iffalse
 %<*driver>
@@ -123,7 +123,7 @@
 %
 % The package accepts two further options.
 % \begin{description}
-% \item[final] to supress the action of this package, for `final'
+% \item[final] to suporess the action of this package, for `final'
 % versions.
 % \item[draft] the normal behaviour of this package.
 % \end{description}
@@ -179,7 +179,7 @@
 % Displayed math (without |equation| counter).
 % \[0=0\label{disp}\]
 %
-% Some text refering to the maths on page~\pageref{disp},  and the
+% Some text referring to the maths on page~\pageref{disp},  and the
 % item~\ref{e^1}.\footnotetext{Actually \texttt{\string\marginpar} is
 % not used at all in this package now.}
 %
@@ -234,10 +234,10 @@
 % \bibitem{GN:sl}
 % Gil Neiger, \emph{showlabels.sty},
 % Undated package, similar to this one, but shows labels
-% inline, affecting the formating of the document.
+% inline, affecting the formatting of the document.
 %
 % \bibitem{anon:sk}
-% Annonymous, \emph{showkeys.sty},
+% Anonymous, \emph{showkeys.sty},
 % Package, dated 14 May 1988. Very similar to this one,
 % also uses |\marginpar| in outer vertical mode.
 %
@@ -257,7 +257,7 @@
 %    \end{macrocode}
 %
 %
-% First we handle the options. Normally all related comands are
+% First we handle the options. Normally all related commands are
 % defined to show their `keys'. But since v3.03 one can specify:
 %
 % \texttt{notref} to stop the redefinition of |\ref| (and |\pageref|,
@@ -302,7 +302,7 @@
 % \changes{v3.15}{2007/08/07}
 %      {Fix \cs{showkeyslabelformat} for final option PR/3918.}
 % Allow |final| to be specified in the document class options
-% to supress the loading of this package.
+% to suppress the loading of this package.
 %    \begin{macrocode}
 \DeclareOption{final}{%
   \providecommand*\showkeyslabelformat[1]{}%
@@ -408,7 +408,7 @@
 % If the |\label| is straight after |\item| (|\bibitem| is handled by
 % this case as well) then the item label has not been added to the page
 % yet. It is hanging around in the box |\@labels| waiting for the
-% paragraph to start. So just need to attatch the label to this box.
+% paragraph to start. So just need to attach the label to this box.
 %    \begin{macrocode}
       \global\setbox\@labels\hbox{%
         \llap{\SK@lab\SK@lab@relax
@@ -422,7 +422,7 @@
 % \changes{v3.10}{1996/09/06}
 %      {Save prevdepth and restore later}
 % If we insert a box into the main vertical list, do not want to
-% change |\prevdepth| as that would afect vertical spacing in the
+% change |\prevdepth| as that would affect vertical spacing in the
 % document. (The box itself should not cause any difference in break
 % points as there is a node there anyway coming from the |\write| to
 % the aux file.
@@ -833,7 +833,7 @@
 % \changes{v3.12}{1997/06/12}
 %         {Support cite package. tools/2490}
 % \changes{v3.17}{2014/04/24}
-%         {rededine \cs{@citex} rather than \cs{cite} for tools/4162}
+%         {redefine \cs{@citex} rather than \cs{cite} for tools/4162}
 %
 % If \textsf{cite} or \textsf{overcite} is being used, redefine |\citen|
 % rather than |\cite| so as not to spoil the space and punctuation
@@ -868,7 +868,7 @@
 %    \end{macrocode}
 %
 % \changes{v3.17}{2014/04/24}
-%         {Change name, rededine \cs{@citex} rather than \cs{cite} for tools/4162}
+%         {Change name, redefine \cs{@citex} rather than \cs{cite} for tools/4162}
 %    \begin{macrocode}
 \def\SK@citex[#1]#2{%
   \SK@\SK@@ref{#2}\SK@@citex[{#1}]{#2}}
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/tabularx.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/tabularx.dtx
@@ -1,7 +1,7 @@
 % \iffalse meta-comment
 %
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009 2014
+% Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -31,7 +31,7 @@
 %<driver> \ProvidesFile{tabularx.drv}
 % \fi
 %         \ProvidesFile{tabularx.dtx}
-          [2014/05/13 v2.10 `tabularx' package (DPC)]
+          [2014/10/28 v2.10 `tabularx' package (DPC)]
 % \iffalse
 %<*driver>
 \documentclass{ltxdoc}
@@ -301,7 +301,7 @@
 % \begin{macro}{\tabularx}
 % Prior to v1.06, this macro took two arguments, which were saved in
 % separate registers before the table body was saved by "\TX@get@body".
-% Unfortunatly this disables the "[t]" optional argument. Now just save
+% Unfortunately this disables the "[t]" optional argument. Now just save
 % the width specification separately, then clear the token register
 % "\toks@".
 % Finally call "\TX@get@body" to begin saving the body of the table.
@@ -710,7 +710,7 @@
    \expandafter\let\expandafter\tabularx\csname tabular*\endcsname
    \expandafter\let\expandafter\endtabularx\csname endtabular*\endcsname
 %    \end{macrocode}
-% Added at v1.05: dissable "\write"s during a trial run. This trick is
+% Added at v1.05: disable "\write"s during a trial run. This trick is
 % from the \TeX{}Book.\footnote{Actually the \TeX{}Book trick does
 % not work correctly, so changed for v2.05.}
 % \changes{v2.05}{1997/09/18}
@@ -767,7 +767,7 @@
 %
 % \begin{macro}{\arraybackslash}
 % "\\" hack.
-% \changes{v2.08}{2012/02/06}{Dont define this if already defined by newer array package and use tabularnewline now this is in 2e.}
+% \changes{v2.08}{2012/02/06}{Don't define this if already defined by newer array package and use tabularnewline now this is in 2e.}
 %    \begin{macrocode}
 \ifx\arraybackslash\@undefined
 \def\arraybackslash{\let\\\tabularnewline}
@@ -822,7 +822,7 @@
 % \end{macro}
 %
 % This last section was added at Version 1.02. Previous versions
-% documentented the fact that "\verb" did not work inside {\ttfamily
+% documented the fact that "\verb" did not work inside {\ttfamily
 % tabularx}, but that did not stop people using it! This usually put
 % \LaTeX\ into an irrecoverable error position, with error messages that
 % did not mention the cause of the error. The `poor man's "\verb"' (and
@@ -837,7 +837,7 @@
 % "\let\verb=\TX@verb". (Making sure to restore the real definition
 % later!)
 %
-% "\verb" and "\verb*" are subject to the following restictions:
+% "\verb" and "\verb*" are subject to the following restrictions:
 % \begin{enumerate}
 % \item Spaces in the argument are not read verbatim, but may be skipped
 %       according to \TeX's usual rules.
@@ -927,7 +927,7 @@
 %
 % \begin{macro}{\TX@v@}
 % Loop through the "\meaning", replacing all spaces by "~". If the last
-% charcter is a space it is dropped, so that "\verb*|\LaTeX|" produces
+% character is a space it is dropped, so that "\verb*|\LaTeX|" produces
 % "\LaTeX" not \verb*|\LaTeX |. The rewritten tokens are then further
 % processed to replace "##" pairs.
 %    \begin{macrocode}
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/theorem.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/theorem.dtx
@@ -1,7 +1,7 @@
 % \iffalse meta-comment
 %
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009
+% Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -40,7 +40,7 @@
 %%
 %
 %
-\def\FMithmInfo{1995/11/23 v2.2c Theorem extension package (FMi)}
+\def\FMithmInfo{2014/10/28 v2.2c Theorem extension package (FMi)}
 %
 % \ProvidesFile{theorem.dtx}[\FMithmInfo]
 %\iffalse   % this is a METACOMMENT !
@@ -340,7 +340,7 @@
 % \begin{verbatim}
 %      \theoremheaderfont{\normalfont\bfseries}
 %\end{verbatim}
-% i.e.\ supplying all necessary font informations explicitly.
+% i.e.\ supplying all necessary font information explicitly.
 %
 %
 %
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/trace.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/trace.dtx
@@ -1,6 +1,7 @@
 % \iffalse meta-comment
 %
 % Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -34,7 +35,7 @@
 %<driver>\ProvidesFile{trace.drv}
 % \fi
 %         \ProvidesFile{trace.dtx}
-          [2014/04/21 v1.1d trace LaTeX code]
+          [2014/10/28 v1.1d trace LaTeX code]
 %
 % \iffalse
 %<*driver>
@@ -316,7 +317,7 @@
 %    |\conditionally@traceon| should do nothing. We set them to
 %    |\@empty| not |\relax| since the latter might produce a math ord
 %    in certain circumstances.  We also have to make sure that
-%    |\traceon| (as defined further down) is not redfining
+%    |\traceon| (as defined further down) is not redefining
 %    |\conditionally@traceoff| again. To make this all work these
 %    redefinitions have to wait until the end of the package.
 %  \changes{v1.1c}{2003/04/30}{Option ``full'' added}
@@ -436,7 +437,7 @@
 %    it revert all settings changed by |\tr@ce@n|? It should not, since
 %    our goal is to shorten the trace output, thus setting all of the
 %    uninteresting values back makes the output unnecessarily
-%    longer. Therefore we restrict ourself to those |\tracing...|
+%    longer. Therefore we restrict ourselves to those |\tracing...|
 %    internals that really contribute to listings like the above.
 %
 %    And one additional point is worth mentioning. The order in which
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/varioref.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/varioref.dtx
@@ -1,6 +1,7 @@
 % \iffalse meta-comment
 %
 % Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -33,7 +34,7 @@
 %
 %<package>\NeedsTeXFormat{LaTeX2e}[1995/05/16]
 %<package>\ProvidesPackage{varioref}
-%<package>    [2014/07/03 v1.5a package for extended references (FMi)]
+%<package>    [2014/10/28 v1.5a package for extended references (FMi)]
 % \fi
 %
 % \CheckSum{2165}
@@ -84,7 +85,7 @@
 %
 % \section{Introduction}
 %
-% In many cases it is helpful when refering to a figure or table to
+% In many cases it is helpful when referring to a figure or table to
 % put both a |\ref| and a |\pageref| command into the document
 % especially when there are one or more pages between the reference
 % and the object.  Therefore some people use a command like
@@ -114,7 +115,7 @@
 %
 %\DescribeMacro\vpageref Sometimes one wants to refer only to page
 % number and again such a reference should normally be suppressed if
-% we are refering to the current page. For this purpose the package
+% we are referring to the current page. For this purpose the package
 % defines the |\vpageref| command. It will produce the same strings as
 % |\vref| except that it doesn't start with the |\ref| and except that
 % it will produce the string that is saved in |\reftextcurrent| if
@@ -323,7 +324,7 @@
 %\DescribeMacro\reftextfaceafter Backward references use
 % |\reftextbefore| if the label is on the preceding page but invisible
 % and |\reftextfacebefore| if it is one the facing page (i.e., if the
-% current page number is odd).  Similarily |\reftextafter| is used
+% current page number is odd).  Similarly |\reftextafter| is used
 % when the label comes on the next page but one has to turn the page
 % and |\reftextfaceafter| if it is on the following but facing page.
 %
@@ -1022,9 +1023,9 @@
     \def\reftextlabelrange#1#2{\ref{#1} ws\nobreakspace\ref{#2}}%
   }}
 %    \end{macrocode}
-%    Iselandic defaults suggested by Thorhallur Sverrisson
+%    Icelandic defaults suggested by Thorhallur Sverrisson
 %    (\verb=thorhs@basis.is=).
-% \changes{v1.4t}{2007/09/07}{Added defaults for islandic}
+% \changes{v1.4t}{2007/09/07}{Added defaults for icelandic}
 %    \begin{macrocode}
 \DeclareOption{icelandic}
   {\vref@addto\extrasicelandic{%
@@ -1233,7 +1234,7 @@
 %    I.~A.~Kotelnikov and P.~Z.~Chebotaev, and adapted
 %    to the encoding-independent form by Vladimir Volovich
 %    (\verb=vvv@vsu.ru=).
-% \changes{v1.2a}{1998/11/23}{Added russion strings}
+% \changes{v1.2a}{1998/11/23}{Added russian strings}
 % \changes{v1.2a}{1998/11/24}{Added defaults for
 %      \cs{reftextpagerange} and \cs{reftextlabelrange}}
 %    \begin{macrocode}
@@ -1584,7 +1585,7 @@
 %    stream.  Otherwise we insert the second argument.  Recall that
 %    those are the two optional arguments of |...|. However, if the
 %    user explicitly requested an empty text string using the optional
-%    argument we cancel any preceeding space.
+%    argument we cancel any preceding space.
 % \changes{v1.2f}{2000/07/07}{Don't output space if optional argument
 %    empty (pr/3230)}
 %    \begin{macrocode}
@@ -1622,7 +1623,7 @@
 %    pages. |\thevpagerefnum| will expand to the page number of the object
 %    (and we know that this is a number) and |\@tempcnta| is either
 %    one higher than the reference page or completely out of bounds.
-%    So if both represend the same value then the object lies one page
+%    So if both represent the same value then the object lies one page
 %    after its reference.
 %    \begin{macrocode}
         \ifnum \thevpagerefnum =\@tempcnta
@@ -1659,9 +1660,9 @@
 %    does not occur, but of course it isn't impossible if you more
 %    than one reference to the same object, or if you have back
 %    references to sections, theorems, etc.  To test this we now
-%    substract two from the current value of |\@tempcnta| (which was
+%    subtract two from the current value of |\@tempcnta| (which was
 %    set to one higher as the reference page number). Note, that
-%    substraction is also possible if the value was |\maxdimen| --- we
+%    subtraction is also possible if the value was |\maxdimen| --- we
 %    still get something that is much larger than any sensible page
 %    number.
 %    \begin{macrocode}
@@ -1732,7 +1733,7 @@
 % \changes{v1.4e}{2003/01/08}{Macro added (pr/3489)}
 %    This macro is like |\label| in the kernel but without passing on
 %    |\@currentlabel|, ie useful if all we are interested in is the
-%    page number. It also has the effect that it is uneffected by
+%    page number. It also has the effect that it is unaffected by
 %    \texttt{amsmath} or Babel redefining the |\label| command.
 %    \begin{macrocode}
 \def\vref@label#1{\@bsphack
@@ -1807,7 +1808,7 @@
 % \changes{v1.2a}{1998/11/27}{Add \cs{leavevmode}}
 %
 %    Above comments all refer to the earlier versions where there was
-%    no posibility to suppress the space at all. By now we have
+%    no possibility to suppress the space at all. By now we have
 %    |\vref*| and |\vpageref*| to suppress the in internally generated
 %    space by redefining |\vref@space| appropriately.
 %    \begin{macrocode}
@@ -1911,7 +1912,7 @@
 \def\vref@pagenum#1#2{%
   \@ifundefined{r@#2}{\@namedef{r@#2}{{??}{??}}}{}%
 %    \end{macrocode}
-%    Extracting and saving the page number turned out to be a soure of subtle
+%    Extracting and saving the page number turned out to be a source of subtle
 %    bugs. Initially it was done through an |\edef| with a bunch of
 %    |\expandafter| commands inside. This posed a problem if the page number
 %    itself contained code which needed protection (e.g., pr/4080) so this got
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/verbatim.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/verbatim.dtx
@@ -1,7 +1,7 @@
 % \iffalse meta-comment
 %
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009 2011
+% Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -512,7 +512,7 @@
 %<*package>
 \NeedsTeXFormat{LaTeX2e}
 \ProvidesPackage{verbatim}
-     [2003/08/22 v1.5q LaTeX2e package for verbatim enhancements]
+     [2014/10/28 v1.5q LaTeX2e package for verbatim enhancements]
 \@ifundefined{verbatim@@@}{}{\endinput}
 %    \end{macrocode}
 %
@@ -1817,7 +1817,7 @@
 Text Text Text Text Text Text Text Text Text Text Text
 Text Text Text Text Text Text Text Text Text Text Text
   \begin {verbatimlisting}{verbtest.tex}
-    Additonal verbatim text
+    Additional verbatim text
       ...
   \end{verbatimlisting}
 And here for listing a file:
--- texlive-base-jessie.git.orig/texmf-dist/source/latex/tools/xspace.dtx
+++ texlive-base-jessie.git/texmf-dist/source/latex/tools/xspace.dtx
@@ -1,7 +1,7 @@
 % \iffalse meta-comment
 %
-% Copyright 1993 1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004 2005
-% 2006 2008 2009
+% Copyright 1993-2014
+%
 % The LaTeX3 Project and any individual authors listed elsewhere
 % in this file.
 %
@@ -33,7 +33,7 @@
 %<driver>\ProvidesFile{xspace.drv}
 % \fi
 %         \ProvidesFile{xspace.dtx}
-          [2009/10/20 v1.13 Space after command names (DPC,MH)]
+          [2014/10/28 v1.13 Space after command names (DPC,MH)]
 %
 % \iffalse
 %<*driver>
@@ -386,7 +386,7 @@
 \def\@xspace{%
 %    \end{macrocode}
 % Before we start checking the exception list it makes sense to
-% performe a quick check on the token in question. Most of the time
+% perform a quick check on the token in question. Most of the time
 % \cs{xspace} is used in regular text so \cs{@let@token} is set equal
 % to a letter. In that case there is no point in checking the list
 % because it will definitely not contain any tokens with catcode~11.
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/afterpage.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/afterpage.sty
@@ -33,7 +33,7 @@
 %% File: afterpage.dtx Copyright (C) 1994 1994 1995 David Carlisle
 \NeedsTeXFormat{LaTeX2e}
 \ProvidesPackage{afterpage}
-         [1995/10/27 v1.08 After-Page Package (DPC)]
+         [2014/10/28 v1.08 After-Page Package (DPC)]
 \newtoks\AP@output
 \global\AP@output\expandafter{\the\output}
 \newbox\AP@partial
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/array.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/array.sty
@@ -34,7 +34,7 @@
 %% Copyright (C) 1989-1998 Frank Mittelbach, all rights reserved.
 \NeedsTeXFormat{LaTeX2e}[1995/06/01]
 \ProvidesPackage{array}
-         [2008/09/09 v2.4c Tabular extension package (FMi)]
+         [2014/10/28 v2.4c Tabular extension package (FMi)]
 %% \CharacterTable
 %%  {Upper-case    \A\B\C\D\E\F\G\H\I\J\K\L\M\N\O\P\Q\R\S\T\U\V\W\X\Y\Z
 %%   Lower-case    \a\b\c\d\e\f\g\h\i\j\k\l\m\n\o\p\q\r\s\t\u\v\w\x\y\z
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/bm.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/bm.sty
@@ -36,7 +36,7 @@
 %% http://www.yandy.com
 \NeedsTeXFormat{LaTeX2e}
 \ProvidesPackage{bm}
-          [2004/02/26 v1.1c Bold Symbol Support (DPC/FMi)]
+          [2014/10/28 v1.1c Bold Symbol Support (DPC/FMi)]
 \def\bm#1#2{%
   \let\bm@pmb\install@mathalphabet
   \let\bm@pmb@\getanddefine@fonts
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/calc.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/calc.sty
@@ -42,7 +42,7 @@
 %% the normal LaTeX bug report address.
 \NeedsTeXFormat{LaTeX2e}
 \ProvidesPackage{calc}
-          [2007/08/22 v4.3 Infix arithmetic (KKT,FJ)]
+          [2014/10/28 v4.3 Infix arithmetic (KKT,FJ)]
 \def\calc@assign@generic#1#2#3#4{\let\calc@A#1\let\calc@B#2%
     \calc@open(#4!%
     \global\calc@A\calc@B\endgroup#3\calc@B}
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/dcolumn.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/dcolumn.sty
@@ -33,7 +33,7 @@
 %% File: dcolumn.dtx Copyright (C) 1992-1996 1999-2001 David Carlisle
 \NeedsTeXFormat{LaTeX2e}
 \ProvidesPackage{dcolumn}
-          [2001/05/28 v1.06 decimal alignment package (DPC)]
+          [2014/10/28 v1.06 decimal alignment package (DPC)]
 \RequirePackage{array}
 \def\DC@#1#2#3{%
   \uccode`\~=`#1\relax
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/delarray.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/delarray.sty
@@ -33,7 +33,7 @@
 %% File: delarray.dtx Copyright (C) 1991-1994 David Carlisle
 \NeedsTeXFormat{LaTeX2e}
 \ProvidesPackage{delarray}
-         [1994/03/14 v1.01 array delimiter package (DPC)]
+         [2014/10/28 v1.01 array delimiter package (DPC)]
 \RequirePackage{array}[1994/02/03]
 \def\@tabarray{\@ifnextchar[{\@@array}{\@@array[c]}}
 \def\@@array[#1]{\@ifnextchar\bgroup
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/enumerate.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/enumerate.sty
@@ -33,7 +33,7 @@
 %% File: enumerate.dtx Copyright  1993 1994 1999 David Carlisle
 \NeedsTeXFormat{LaTeX2e}
 \ProvidesPackage{enumerate}
-          [1999/03/05 v3.00 enumerate extensions (DPC)]
+          [2014/10/28 v3.00 enumerate extensions (DPC)]
 \newtoks\@enLab
 \def\@enQmark{?}
 \def\@enLabel#1#2{%
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/ftnright.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/ftnright.sty
@@ -34,7 +34,7 @@
 %% Copyright (C) 1989-2004 Frank Mittelbach, all rights reserved.
 \NeedsTeXFormat{LaTeX2e}[1995/06/01]
 \ProvidesPackage{ftnright}
-         [2010/02/25 v1.1f footnote layout package (FMi)]
+         [2014/10/28 v1.1f footnote layout package (FMi)]
 %%
 %% \CheckSum{431}
 %% \CharacterTable
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/hhline.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/hhline.sty
@@ -33,7 +33,7 @@
 %% File: hhline.dtx Copyright (C) 1991-1994 David Carlisle
 \NeedsTeXFormat{LaTeX2e}
 \ProvidesPackage{hhline}
-         [1994/05/23 v2.03 Table rule package (DPC)]
+         [2014/10/28 v2.03 Table rule package (DPC)]
 \def\HH@box#1#2{\vbox{%
   \hrule \@height \arrayrulewidth \@width #1
   \vskip \doublerulesep
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/layout.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/layout.sty
@@ -32,7 +32,7 @@
 %% 
 \NeedsTeXFormat{LaTeX2e}
 \ProvidesPackage{layout}
-                [2000/09/25 v1.2c Show layout parameters]
+                [2014/10/28 v1.2c Show layout parameters]
 \DeclareOption{dutch}{%
   \def\Headertext{Kopregel}
   \def\Bodytext{Broodtekst}
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/longtable.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/longtable.sty
@@ -33,7 +33,7 @@
 %% File: longtable.dtx Copyright (C) 1990-2001 David Carlisle
 \NeedsTeXFormat{LaTeX2e}[1995/06/01]
 \ProvidesPackage{longtable}
-          [2004/02/01 v4.11 Multi-page Table package (DPC)]
+          [2014/10/28 v4.11 Multi-page Table package (DPC)]
 \def\LT@err{\PackageError{longtable}}
 \def\LT@warn{\PackageWarning{longtable}}
 \def\LT@final@warn{%
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/multicol.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/multicol.sty
@@ -80,7 +80,7 @@
 %%   - multicol is used for non-commercial purposes
 %%
 %%
-%% Determinating a license fee might result in a license fee of zero
+%% Determining a license fee might result in a license fee of zero
 %% (i.e., no payment) in case a producer has determined that the use
 %% of multicol has no enhancing effect on the product. This is a
 %% plausible scenario, i.e., in the above two cases the producer is
@@ -96,8 +96,8 @@
 %%
 \NeedsTeXFormat{LaTeX2e}[1997/12/01]
 \ProvidesPackage{multicol}
-          [2014/08/24 v1.8g  multicolumn formatting (FMi)]
-%% \CheckSum{1840}
+          [2014/10/28 v1.8i  multicolumn formatting (FMi)]
+%% \CheckSum{1853}
 %% \CharacterTable
 %%  {Upper-case    \A\B\C\D\E\F\G\H\I\J\K\L\M\N\O\P\Q\R\S\T\U\V\W\X\Y\Z
 %%   Lower-case    \a\b\c\d\e\f\g\h\i\j\k\l\m\n\o\p\q\r\s\t\u\v\w\x\y\z
@@ -344,6 +344,7 @@
     \fi
   \fi
   \addvspace\multicolsep
+ \prevdepth\z@
   \mult@info\z@
      {Ending environment
              \if@boxedmulticols
@@ -413,6 +414,7 @@
 \def\columnseprulecolor{\normalcolor}
 \def\reinsert@footnotes{\ifvoid\footins\else
          \insert\footins{}\fi}
+\def\vfillmaxdepth{\vskip \z@ \@plus 1fill \@minus \maxdepth}
 \def\multi@column@out{%
    \ifnum\outputpenalty <-\@M
    \speci@ls \else
@@ -445,7 +447,7 @@
                  \vbox to\dimen@
                   {\unvbox\count@
                    \remove@discardable@items
-                   \ifshr@nking\vfill\fi}%
+                   \ifshr@nking\vfillmaxdepth\fi}%
            }%
    \setbox\mult@rightbox
        \vsplit\@cclv to\dimen@
@@ -453,7 +455,7 @@
    \setbox\mult@rightbox\vbox to\dimen@
           {\unvbox\mult@rightbox
            \remove@discardable@items
-           \ifshr@nking\vfill\fi}%
+           \ifshr@nking\vfillmaxdepth\fi}%
    \let\ifshr@king\ifshr@kingsaved
    \ifvoid\@cclv \else
        \unvbox\@cclv
@@ -611,6 +613,7 @@
        \@plus\multicolundershoot
        \@minus\multicolovershoot
    \splitmaxdepth\maxdepth
+   \boxmaxdepth\maxdepth
    \setbox\@tempboxa\vsplit\mult@box to\z@
    \@tempdima\ht\mult@box
    \advance\@tempdima\dp\mult@box
@@ -636,7 +639,7 @@
      \dimen@\topskip
    \fi
    \vbadness\@M
-   \vfuzz \col@number\baselineskip
+   \vfuzz \maxdimen
    \last@try-\p@
    \loop
     {\process@cols\mult@grightbox
@@ -668,7 +671,6 @@
            \too@badtrue
          \fi
                         }}%
-   \boxmaxdepth\maxdepth
    \global\setbox\mult@grightbox
       \vbox{\unvbox\mult@grightbox}%
    \setbox\mult@nat@firstbox
@@ -702,7 +704,7 @@
         \ifnum\badness>\c@finalcolumnbadness
           \global\setbox\mult@grightbox
            \vbox to\dimen@
-              {\unvbox\mult@grightbox\vfill}%
+              {\unvbox\mult@grightbox\vfillmaxdepth}%
            \ifnum\c@tracingmulticols>\@ne
              \message{ setting natural
               (> \the\c@finalcolumnbadness)}%
@@ -746,7 +748,7 @@
               \@plus-\multicolundershoot
               \@minus-\multicolovershoot
             \unvbox\@tempcnta
-            \ifshr@nking\vfill\fi
+            \ifshr@nking\vfillmaxdepth\fi
            }%
         \ifnum\badness>\@M
           \vfuzz\maxdimen % no overfull warning
@@ -888,7 +890,9 @@
    \fi
    \begin{multicols}
 }
- \@namedef{endmulticols*}{\vfill
+ \@namedef{endmulticols*}{\ifshr@nking\else
+                             \vfillmaxdepth\nointerlineskip\null
+                          \fi
    \end{multicols}}
 \mathchardef\@Mv=10005
 \def\columnbreak{%
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/showkeys.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/showkeys.sty
@@ -34,7 +34,7 @@
 %% File: showkeys.dtx Copyright (C) 2006-2014 David Carlisle, LaTeX3 Project
 \NeedsTeXFormat{LaTeX2e}
 \ProvidesPackage{showkeys}
-          [2014/04/24 v3.17 Show cite and label keys (DPC, MH)]
+          [2014/10/28 v3.17 Show cite and label keys (DPC, MH)]
 \DeclareOption{notref}{\let\SK@ref\@empty}
 \DeclareOption{notcite}{\let\SK@cite\@empty}
 \let\SK@refcolor\relax
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/tabularx.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/tabularx.sty
@@ -33,7 +33,7 @@
 %% File: tabularx.dtx Copyright (C) 1991-1999 2012 2014 David Carlisle
 \NeedsTeXFormat{LaTeX2e}
 \ProvidesPackage{tabularx}
-          [2014/05/13 v2.10 `tabularx' package (DPC)]
+          [2014/10/28 v2.10 `tabularx' package (DPC)]
 \DeclareOption{infoshow}{\AtEndOfPackage\tracingtabularx}
 \DeclareOption{debugshow}{\AtEndOfPackage\tracingtabularx}
 \ProcessOptions
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/thb.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/thb.sty
@@ -46,7 +46,7 @@
 %%   Grave accent  \`     Left brace    \{     Vertical bar  \|
 %%   Right brace   \}     Tilde         \~}
 %%
-\def\FMithmInfo{1995/11/23 v2.2c Theorem extension package (FMi)}
+\def\FMithmInfo{2014/10/28 v2.2c Theorem extension package (FMi)}
 %%
 %% Package `theorem' to use with LaTeX2e
 %% Copyright (C) 1989-1995 Frank Mittelbach, all rights reserved.
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/thc.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/thc.sty
@@ -46,7 +46,7 @@
 %%   Grave accent  \`     Left brace    \{     Vertical bar  \|
 %%   Right brace   \}     Tilde         \~}
 %%
-\def\FMithmInfo{1995/11/23 v2.2c Theorem extension package (FMi)}
+\def\FMithmInfo{2014/10/28 v2.2c Theorem extension package (FMi)}
 %%
 %% Package `theorem' to use with LaTeX2e
 %% Copyright (C) 1989-1995 Frank Mittelbach, all rights reserved.
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/thcb.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/thcb.sty
@@ -46,7 +46,7 @@
 %%   Grave accent  \`     Left brace    \{     Vertical bar  \|
 %%   Right brace   \}     Tilde         \~}
 %%
-\def\FMithmInfo{1995/11/23 v2.2c Theorem extension package (FMi)}
+\def\FMithmInfo{2014/10/28 v2.2c Theorem extension package (FMi)}
 %%
 %% Package `theorem' to use with LaTeX2e
 %% Copyright (C) 1989-1995 Frank Mittelbach, all rights reserved.
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/theorem.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/theorem.sty
@@ -46,7 +46,7 @@
 %%   Grave accent  \`     Left brace    \{     Vertical bar  \|
 %%   Right brace   \}     Tilde         \~}
 %%
-\def\FMithmInfo{1995/11/23 v2.2c Theorem extension package (FMi)}
+\def\FMithmInfo{2014/10/28 v2.2c Theorem extension package (FMi)}
 %%
 %% Package `theorem' to use with LaTeX2e
 %% Copyright (C) 1989-1995 Frank Mittelbach, all rights reserved.
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/thm.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/thm.sty
@@ -46,7 +46,7 @@
 %%   Grave accent  \`     Left brace    \{     Vertical bar  \|
 %%   Right brace   \}     Tilde         \~}
 %%
-\def\FMithmInfo{1995/11/23 v2.2c Theorem extension package (FMi)}
+\def\FMithmInfo{2014/10/28 v2.2c Theorem extension package (FMi)}
 %%
 %% Package `theorem' to use with LaTeX2e
 %% Copyright (C) 1989-1995 Frank Mittelbach, all rights reserved.
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/thmb.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/thmb.sty
@@ -46,7 +46,7 @@
 %%   Grave accent  \`     Left brace    \{     Vertical bar  \|
 %%   Right brace   \}     Tilde         \~}
 %%
-\def\FMithmInfo{1995/11/23 v2.2c Theorem extension package (FMi)}
+\def\FMithmInfo{2014/10/28 v2.2c Theorem extension package (FMi)}
 %%
 %% Package `theorem' to use with LaTeX2e
 %% Copyright (C) 1989-1995 Frank Mittelbach, all rights reserved.
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/thp.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/thp.sty
@@ -46,7 +46,7 @@
 %%   Grave accent  \`     Left brace    \{     Vertical bar  \|
 %%   Right brace   \}     Tilde         \~}
 %%
-\def\FMithmInfo{1995/11/23 v2.2c Theorem extension package (FMi)}
+\def\FMithmInfo{2014/10/28 v2.2c Theorem extension package (FMi)}
 %%
 %% Package `theorem' to use with LaTeX2e
 %% Copyright (C) 1989-1995 Frank Mittelbach, all rights reserved.
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/trace.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/trace.sty
@@ -38,7 +38,7 @@
 %%
 \NeedsTeXFormat{LaTeX2e}
 \ProvidesPackage{trace}
-          [2014/04/21 v1.1d trace LaTeX code]
+          [2014/10/28 v1.1d trace LaTeX code]
 \NeedsTeXFormat{LaTeX2e}[1998/12/01]
 \DeclareOption{logonly}
    {\let\tracingonline@p\z@}
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/varioref.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/varioref.sty
@@ -42,7 +42,7 @@
 %%
 \NeedsTeXFormat{LaTeX2e}[1995/05/16]
 \ProvidesPackage{varioref}
-    [2014/07/03 v1.5a package for extended references (FMi)]
+    [2014/10/28 v1.5a package for extended references (FMi)]
 %% \CharacterTable
 %%  {Upper-case    \A\B\C\D\E\F\G\H\I\J\K\L\M\N\O\P\Q\R\S\T\U\V\W\X\Y\Z
 %%   Lower-case    \a\b\c\d\e\f\g\h\i\j\k\l\m\n\o\p\q\r\s\t\u\v\w\x\y\z
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/verbatim.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/verbatim.sty
@@ -34,7 +34,7 @@
 %% Copyright (C) 1989--2003 by Rainer Sch\"opf. All rights reserved.
 \NeedsTeXFormat{LaTeX2e}
 \ProvidesPackage{verbatim}
-     [2003/08/22 v1.5q LaTeX2e package for verbatim enhancements]
+     [2014/10/28 v1.5q LaTeX2e package for verbatim enhancements]
 \@ifundefined{verbatim@@@}{}{\endinput}
 \newtoks\every@verbatim
 \every@verbatim={}
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/verbtest.tex
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/verbtest.tex
@@ -131,7 +131,7 @@
 Text Text Text Text Text Text Text Text Text Text Text
 Text Text Text Text Text Text Text Text Text Text Text
   \begin {verbatimlisting}{verbtest.tex}
-    Additonal verbatim text
+    Additional verbatim text
       ...
   \end{verbatimlisting}
 And here for listing a file:
--- texlive-base-jessie.git.orig/texmf-dist/tex/latex/tools/xspace.sty
+++ texlive-base-jessie.git/texmf-dist/tex/latex/tools/xspace.sty
@@ -35,7 +35,7 @@
 %%                                          Morten H\o gholm
 \NeedsTeXFormat{LaTeX2e}
 \ProvidesPackage{xspace}
-          [2009/10/20 v1.13 Space after command names (DPC,MH)]
+          [2014/10/28 v1.13 Space after command names (DPC,MH)]
 \DeclareRobustCommand\xspace{\@xspace@firsttrue
   \futurelet\@let@token\@xspace}
 \newif\if@xspace@first
--- texlive-base-jessie.git.orig/texmf-dist/tlpkg/tlpobj/tools.doc.tlpobj
+++ texlive-base-jessie.git/texmf-dist/tlpkg/tlpobj/tools.doc.tlpobj
@@ -1,10 +1,10 @@
 name tools.doc
 category Package
-revision 35209
+revision 35577
 shortdesc doc files of tools
 relocated 1
-docfiles size=787
- RELOC/doc/latex/tools/00readme.txt
+docfiles size=793
+ RELOC/doc/latex/tools/README
  RELOC/doc/latex/tools/afterpage.pdf
  RELOC/doc/latex/tools/array.pdf
  RELOC/doc/latex/tools/bm.pdf
--- texlive-base-jessie.git.orig/texmf-dist/tlpkg/tlpobj/tools.source.tlpobj
+++ texlive-base-jessie.git/texmf-dist/tlpkg/tlpobj/tools.source.tlpobj
@@ -1,9 +1,9 @@
 name tools.source
 category Package
-revision 35209
+revision 35577
 shortdesc source files of tools
 relocated 1
-srcfiles size=249
+srcfiles size=250
  RELOC/source/latex/tools/afterpage.dtx
  RELOC/source/latex/tools/afterpage.ins
  RELOC/source/latex/tools/array.dtx
--- texlive-base-jessie.git.orig/texmf-dist/tlpkg/tlpobj/tools.tlpobj
+++ texlive-base-jessie.git/texmf-dist/tlpkg/tlpobj/tools.tlpobj
@@ -1,6 +1,6 @@
 name tools
 category Package
-revision 35209
+revision 35577
 shortdesc The LaTeX standard tools bundle.
 relocated 1
 longdesc A collection of (variously) simple tools provided as part of
@@ -50,5 +50,5 @@
  RELOC/tex/latex/tools/xr.sty
  RELOC/tex/latex/tools/xspace.sty
 catalogue-ctan /macros/latex/required/tools
-catalogue-date 2014-09-06 19:55:20 +0200
+catalogue-date 2014-10-31 15:58:02 +0100
 catalogue-license lppl1.3
--- texlive-base-jessie.git.orig/tlpkg/texlive.tlpdb
+++ texlive-base-jessie.git/tlpkg/texlive.tlpdb
@@ -209069,7 +209069,7 @@
 doccontainersize 2820248
 doccontainermd5 47c0cd51463f5bedd3b367e8c658fbe4
 docfiles size=787
- RELOC/doc/latex/tools/00readme.txt details="Readme"
+ RELOC/doc/latex/tools/README details="Readme"
  RELOC/doc/latex/tools/afterpage.pdf
  RELOC/doc/latex/tools/array.pdf
  RELOC/doc/latex/tools/bm.pdf
